fabric-render-data-v2/src/main/java/net/fabricmc/fabric/api/renderdata/v2/RenderDataBlockEntity.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.renderdata.v2;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Extensions that allow {@link BlockEntity} subclasses to provide render data.
+ *
+ * <p>Block entity render data is arbitrary data that is safe to access and use in a client-side
```

Comment:
I think this should explain that it is retrieved from the BlockRenderView, and why the BE shouldn't be used directly (it might be modified concurrently on the client side).

Reply:
Maybe something like `Directly accessing the fields in a block entity is not safe in a multi-threaded environment, such as chunk building, as the block entity might be modified concurrently on the main thread. Instead, the block entity should first provide the relevant data in getRenderData, and the code that needs this data can then query it from the BlockRenderView.` in the very first paragraph.

Reply:
Ah I see that there's some explanation in `RenderDataBlockView` already. Still, it would be good to have something here too.